crypto/s2n_pkey.c

```diff
@@ -154,11 +141,26 @@ int s2n_pkey_match(const struct s2n_pkey *pub_key, const struct s2n_pkey *priv_k
             break;
         case S2N_PKEY_TYPE_MLDSA:
             check_alg = S2N_SIGNATURE_MLDSA;
+            hash_alg = S2N_HASH_SHAKE256_64;
             break;
         default:
             POSIX_BAIL(S2N_ERR_CERT_TYPE_UNSUPPORTED);
     }
 
+    DEFER_CLEANUP(struct s2n_hash_state state_in = { 0 }, s2n_hash_free);
+    POSIX_GUARD(s2n_hash_new(&state_in));
+    POSIX_GUARD(s2n_hash_init(&state_in, hash_alg));
+    POSIX_GUARD_RESULT(s2n_pkey_init_hash(pub_key, check_alg, &state_in));
```

Comment:
Isn't `s2n_pkey_match` only called when a certificate is loaded into s2n-tls? Why doesn't `s2n_mldsa_init_mu_hash ` need to be called in other circumstances as well?

Reply:
This PR just adds the crypto logic to make signing possible, but doesn't actually allow ML-DSA signatures to be negotiated or used during the TLS handshake. My next PR will create an s2n_signature_scheme to enable ML-DSA signing during the handshake and update our CertVerify logic to call s2n_mldsa_init_mu_hash.

I wonder if I should name this PR something else? :thinking: